DDD Workshop.ipynb

```diff
@@ -0,0 +1,547 @@
+    "\n",
+    "Let's start with a look at the occupancy rates for all of the specific flights, that is every combo of airport and time:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "LAX_WEEKDAY\n",
+      " 0.8508\n",
+      "JFK_WEEKDAY\n",
+      " 0.83\n",
+      "YVR_WEEKDAY\n",
+      " 0.8165\n",
+      "LAX_REDEYE\n",
+      " 0.8531\n",
+      "JFK_REDEYE\n",
+      " 0.8807\n",
+      "YVR_REDEYE\n",
+      " 0.8192\n",
+      "LAX_WEEKEND\n",
+      " 0.8508\n",
+      "JFK_WEEKEND\n",
+      " 0.8307\n",
+      "YVR_WEEKEND\n",
+      " 0.8666\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "[None, None, None, None, None, None, None, None, None]"
+      ]
+     },
+     "execution_count": 3,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "cleaned_data_file = open(\"cleaned_flight_data.csv\", \"r\")\n",
+    "\n",
+    "data_by_airport_time = {}\n",
+    "for line in cleaned_data_file:\n",
+    "    fields = line.split(\",\")\n",
+    "    airport = fields[0]\n",
+    "    if airport == \"AIRPORT\":\n",
+    "        continue\n",
+    "    time = fields[3]\n",
+    "    \n",
+    "    key = \"{}_{}\".format(airport, time)\n",
+    "    if key not in data_by_airport_time.keys():\n",
+    "        data_by_airport_time[key] = []\n",
+    "    \n",
+    "    occupancy_percent = int(fields[2]) / int(fields[1])\n",
+    "    data_by_airport_time[key].append(occupancy_percent)\n",
+    "    \n",
+    "cleaned_data_file.close()\n",
+    "\n",
+    "rate_by_airport_time = {}\n",
+    "for key, data_arr in data_by_airport_time.items():\n",
+    "    occupancy_average = sum(data_arr) / len(data_arr)\n",
+    "    rate_by_airport_time[key] = occupancy_average\n",
+    "\n",
+    "[print(k, round(v, 4)) for k,v in rate_by_airport_time.items()]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "There we go! We can now make a conclusion about which flights would be best to add to the current schedule. From the looks of it, another redeye to JFK would be the best bet here, but this relies on the assumption that the additional supply would produce additional demand. If the demand is static for redeye flights, meaning the supply changing has little effect on the demand, then adding flight here would do very little. It is also possible that the current flights, even though they are the most full, do not have any more demand for additional seats. These aren't questions we can answer with this dataset alone but might be one you could get from somebody else working at the airport.\n",
```

Comment:
Reword last sentence here to: These aren't questions we can answer with this dataset alone, but maybe you could get some answers from somebody else working at the airport.